src/utils/MultiInviter.js

```diff
@@ -85,18 +92,23 @@ export default class MultiInviter {
     }
 
     getErrorText(addr) {
-        return this.errorTexts[addr];
+        return this.errors[addr] ? this.errors[addr].errorText : null;
     }
 
-    async _inviteToRoom(roomId, addr) {
+    async _inviteToRoom(roomId, addr, ignoreProfile) {
         const addrType = getAddressType(addr);
 
         if (addrType === 'email') {
             return MatrixClientPeg.get().inviteByEmail(roomId, addr);
         } else if (addrType === 'mx-user-id') {
-            const profile = await MatrixClientPeg.get().getProfileInfo(addr);
-            if (!profile) {
-                return Promise.reject({errcode: "M_NOT_FOUND", error: "User does not have a profile."});
+            if (!ignoreProfile && !SettingsStore.getValue("alwaysRetryInvites", this.roomId)) {
+                const profile = await MatrixClientPeg.get().getProfileInfo(addr);
+                if (!profile) {
+                    return Promise.reject({
```

Comment:
isn't this just 'throw'?

Also I find faking up errors in the client that look like server errors can be quite confusing: I would probably throw a specific exception type here.